### PR TITLE
Pipeline LSP FS requests for batch_open

### PR DESCRIPTION
**Pull request by keegancsmith**
This is a perf improvement for workspace/symbols. Previously we would send a
request and wait for a response for a file in serial. On my laptop
workspace/symbols for django goes from 4.5s to 3s.

### CODE REVIEW

File: langserver/jsonrpc.py [hunk 1]
```diff
@@ -136,6 +138,41 @@ def send_request(self, method: str, params):
         self._send(body)
         return self.read_message(want=lambda msg: msg.get("id") == rid)
 
+    def send_request_batch(self, requests):
+        """Pipelines requests and returns responses.
+
+        The responses is a generator were the nth response corresponds with the
```
Comment (beyang) [inline langserver/jsonrpc.py]: s/were/where/

**Comment by beyang** [comment]
@keegancsmith Looks good to me, but I don't see where `batch_open` is called. Is that changed in another changeset?

**Comment by beyang** [comment]
LGTM

**Comment by keegancsmith** [comment]
> @keegancsmith Looks good to me, but I don't see where batch_open is called. Is that changed in another changeset?

https://github.com/sourcegraph/python-langserver/blob/43142ee0c3c62660807326f28960bee25fa4f99e/langserver/langserver.py#L104

Maybe when we have python support on sg.com you would of more easily found that :P

**Pull request merged**